### PR TITLE
[FAU-434] Correctly assign campo keys to multiple terms within degree program

### PR DESCRIPTION
**Pull request by zhyian**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)

https://inpsyde.atlassian.net/browse/FAU-434

Currently we have only one campo key per taxonomy in the degree program data, even if degree program has a few terms with  different campo keys. 

**What is the new behavior (if this is a feature change)?**

The new logic presented in related [PR](https://github.com/RRZE-Webteam/FAU-Studium-Common/pull/17), includes campo keys from all related terms to degree program data. And in this PR, we update the current implementation to handle an array of campo keys instead of a single string, as was previously done.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:
https://github.com/RRZE-Webteam/FAU-Studium-Common/pull/17

### CODE REVIEW

**Review by tyrann0us — COMMENTED** [review]
Thanks! Could you please check the Psalm errors?

**Review by tyrann0us — APPROVED** [review]
> Psalm errors are occurring because the related PR ([RRZE-Webteam/FAU-Studium-Common#17](https://github.com/RRZE-Webteam/FAU-Studium-Common/pull/17)) in the common package hasn't been merged into the dev branch yet.

I also thought of that but didn't realize that was indeed the case. Sorry for the noise!

File: src/Infrastructure/Search/FilterableTermsUpdater.php [hunk 1]
```diff
@@ -540,16 +542,25 @@ private static function arraySearchBy(array $array, string $property, mixed $val
...
+            $campoKeys = $rawView->campoKeys()->asArray();
+            $campoKeyType = CampoKeysRepository::TAXONOMY_TO_CAMPO_KEY_MAP[$taxonomy] ?? '';
+            $taxonomyCampoKeys = $campoKeys[$campoKeyType] ?? null;
 
-        if (is_null($campoKey)) {
-            return;
-        }
+            if (!is_array($taxonomyCampoKeys)) {
+                continue;
+            }
```
Comment (shvlv) [inline src/Infrastructure/Search/FilterableTermsUpdater.php:555]: It looks like it might be checked out of the loop.

**Review by shvlv — APPROVED** [review]

**Review by amiut — APPROVED** [review]
thank you

**Pull request merged**